starter-code/routes/passportRouter.js

```diff
@@ -1,22 +1,70 @@
-const express        = require("express");
-const router         = express.Router();
+const express = require("express");
+const router = express.Router();
 // User model
-const User           = require("../models/user");
+const User = require("../models/user");
 // Bcrypt to encrypt passwords
-const bcrypt         = require("bcrypt");
-const bcryptSalt     = 10;
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
 const ensureLogin = require("connect-ensure-login");
-const passport      = require("passport");
+const passport = require("passport");
 
 
+router.get("/login", (req, res, next) => {
+    res.render("passport/login", { "message": req.flash("error") });
+});
+
+router.post("/login", passport.authenticate("local", {
+    successRedirect: "/private",
+    failureRedirect: "/login",
+    failureFlash: true,
+    passReqToCallback: true
+}));
+
 
-router.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
-  res.render("passport/private", { user: req.user });
+router.get("/signup", (req, res, next) => {
+    res.render("passport/signup");
 });
 
+router.post("/signup", (req, res, next) => {
+    const username = req.body.username;
+    const password = req.body.password;
```

Comment:
Little ES6 trick: you can do that in one line with `const { username, password } = req.body`.